pkg/cosign/kubernetes/client.go

```diff
@@ -57,3 +58,16 @@ func Client() (kubernetes.Interface, error) {
 	}
 	return kubernetes.NewForConfig(config)
 }
+
+func CheckImmutableSecretSupported(client kubernetes.Interface) (bool, error) {
```

Comment:
nit: It is been used internally in the package, `func checkImm....`.

Reply:
+1 here, otherwise LGTM!